grpc/pb/space.proto

```diff
@@ -95,4 +106,66 @@ message CreateFolderRequest {
 }
 // not sure we need to return anything other than an error if we failed
 message CreateFolderResponse {
+}
+
+message GetIdentityByUsernameRequest {
+  string username = 1;
+}
+
+message Identity {
+  string address = 1;
+  string publicKey = 2;
+  string username = 3;
+}
+
+message GetIdentityByUsernameResponse {
+  Identity identity = 1;
+}
+
+message CreateUsernameAndEmailRequest {
+  string username = 1;
+  string email = 2; // Optional. Send "" to skip.
+}
+
+message CreateUsernameAndEmailResponse {}
+
+message BackupKeysByPassphraseRequest {
+  string passphrase = 1;
+}
+
+message BackupKeysByPassphraseResponse {}
+
+message RecoverKeysByPassphraseRequest {
+  string passphrase = 1;
+}
+
+message RecoverKeysByPassphraseResponse {}
+
+message ShareBucketViaEmailRequest {
+  string bucket = 1;
+  string email = 2;
+}
+
+message ShareBucketViaEmailResponse {}
+
+enum IdentityType {
+  USERNAME = 0;
+  EMAIL = 1;
```

Comment:
Does email take precedence if they have both?

Reply:
This is just for the `ShareBucketViaIdentityRequest` method. So you can use this to either share a bucket to a user through username or through email (e.g. if I can send `ShareBucketViaIdentityRequest(USERNAME, "jsonsivar", "my-bucket")`). I guess we can add ADDRESS as an enum field too if we need it.

Reply:
> LGTM just had some clarifications.
> 
> Here are some other interfaces that come to mind. If you think they make sense, I can create a PR targeting your branch lmk:
> 
> Login or user initialization (either via private key, password or email depending on what we finalize)

That's partially handled by `RecoverKeysByPassphrase`. We can add another one to import private key too I think.

> Logout or unset user

Makes sense. But I'd call this `DeleteUserKeys` or something like that because that's what would be actually happening.

> Create team (based on a list of identities - maybe we can leave an optional field for read vs write which we can use later)

Makes sense, not too sure how that's handled by the daemon yet tho.

> Create shared bucket(bucket name, team id)

Or maybe just create bucket with a param to set if you want it shared or not. Maybe it can be part of the UI (e.g. there's a toggle next to each bucket to choose on/off for "automatic back up")

> Get team members (for now to use our own db and can upgrade to Textile when this gets added)

Sounds good

> Remove member (team id, identity)

Sounds good too. Maybe Add Member too?